frontend/packages/dev-console/src/components/pipelines/modals/triggers/AddTriggerModal.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from 'react';
+import { Formik } from 'formik';
+import {
+  createModalLauncher,
+  ModalComponentProps,
+} from '@console/internal/components/factory/modal';
+import { Pipeline } from '../../../../utils/pipeline-augment';
+import ModalStructure from '../common/ModalStructure';
+import { convertPipelineToModalData } from '../common/utils';
+import { addTriggerSchema } from '../common/validation-utils';
+import AddTriggerForm from './AddTriggerForm';
+import { TRIGGER_BINDING_EMPTY } from './const';
+import { submitTrigger } from './submit-utils';
+import { AddTriggerFormValues } from './types';
+
+type AddTriggerModalProps = ModalComponentProps & {
+  pipeline: Pipeline;
+};
+
+const AddTriggerModal: React.FC<AddTriggerModalProps> = ({ pipeline, close }) => {
+  const initialValues: AddTriggerFormValues = {
+    ...convertPipelineToModalData(pipeline, true),
+    triggerBinding: {
+      name: TRIGGER_BINDING_EMPTY,
+      resource: null,
```

Comment:
Why do we have to pass null here?

Reply:
No selected [k8s] resource at form startup -- it's a required field by the types so nothing is hidden.